src/driver/verify.cpp

```diff
@@ -237,6 +238,57 @@ void verify_reduced_program(const program& p,
     }
 }
 
+bool verify_bisect(program p,
+                    int mid_n,
+                    const target& t,
+                    compile_options options,
+                    verify_options vo,
+                    const parameter_map& inputs,
+                    verify::tolerance tols)
+{
+    auto* mm  = p.get_main_module();
+    auto mid = std::next(mm->begin(), mid_n);
+    mm->remove_instructions(mid, mm->end());
+    std::cout << "Verify up to: " << mid_n << std::endl;
+    std::cout << p << std::endl;
+    try
+    {
+        return verify_program(std::to_string(mid_n), p, t, options, vo, inputs, tols);
+    }
+    catch(const std::exception& e)
+    {
+        std::cout << "FAILED: " << mid_n << std::endl;
+        std::cout << "Exception: " << e.what() << std::endl;
+        return false;
+    }
+
+}   
+
+void verify_bisected_program(const program& p,
+                            const target& t,
+                            compile_options options,
+                            verify_options vo,
+                            const parameter_map& inputs,
+                            verify::tolerance tols)
+{
+    const auto* mm = p.get_main_module();
+    auto right         = std::distance(mm->begin(), mm->end());
+    std::size_t left = 0;
+    bool passed;
+
+    std::cout << "Bisect Verify steps: " << right << std::endl;
+    while (left <= right) {
+        std::size_t mid = left + (right - left) / 2;
```

Comment:
You need to skip the instructions that are const foldable.

Reply:
You can probably create a vector to do the mapping:

```cpp
std::vector<std::size_t> trim_map;
std::size_t i = 0;
for(const auto& ins:*mm)
{
    if(not ins.can_eval())
        trim_map.push_back(i);
    i++;
}
```

Then do the bisection over `trim_map` and then use it to get the actual trim value: `verify_bisect(p, trim_map[mid], t, options, vo, inputs, tols)`.